openedx/features/course_duration_limits/config.py

```diff
@@ -50,3 +51,8 @@ def set_value_for_content_type_gating_holdback(sender, event=None, user=None, **
                 )
             except (ExperimentKeyValue.DoesNotExist, AttributeError):
                 pass
+            except IntegrityError:
+                # There is a race condition when multiple enrollments happen at the same time where the ExperimentData
```

Comment:
have we verified this is actually the case?  
I imagine this signal would fire much quicker than a user can navigate through the UI to a second course and enroll.  
perhaps this triggers in some sort of special case like a script or some bundles.  
I suppose maybe we don't care why we get the IntegrityError if it doesn't happen very frequently.  
Can we see how often it happens?

Reply:
Looking through the logs there were multiple users who were able to register for multiple courses in the same second. There's were only a handful of cases of this error since we turned it on and they all seem to be related to multiple simultaneous registrations.